content/5_Plat/2DRQ.mdx

```diff
@@ -207,11 +208,137 @@ factors for both are terrible.
 
 This doesn't require knowing the queries beforehand.
 
-- [my 1D offline BIT](<https://github.com/bqi343/USACO/blob/master/Implementations/content/data-structures/1D%20Range%20Queries%20(9.2)/BIToff.h>)
-- [my 2D offline BIT](<https://github.com/bqi343/USACO/blob/master/Implementations/content/data-structures/2D%20Range%20Queries%20(15.2)/BIT2DOff%20(15.2).h>)
+- [Benq's 1D offline BIT](<https://github.com/bqi343/USACO/blob/master/Implementations/content/data-structures/1D%20Range%20Queries%20(9.2)/BIToff.h>)
+- [Benq's 2D offline BIT](<https://github.com/bqi343/USACO/blob/master/Implementations/content/data-structures/2D%20Range%20Queries%20(15.2)/BIT2DOff%20(15.2).h>)
 
-It's a bit difficult to pass the above problem within the time limit. Make sure
-to use fast input (and not `endl`)!
+Here's an implementation of the offline 2D BIT presented above that may be
+easier to understand, albeit significantly slower (same complexity, just worse constant factor):
+
+<LanguageSection>
+<CPPSection>
+```cpp
+// see code example below for usage
+struct BIT2D {
+	bool mode = 0;  // 0 -> offline, 1 -> initialized
+	vector<pair<int, int>> todo;
+	vector<int> vals[N], bit[N];
+	/** initializes BIT after all offline updates */
+	void init() {
+		assert(!mode);
+		// sort points by y-coordinate
+		sort(todo.begin(), todo.end(),
+		     [](P a, P b) { return a.second < b.second; });
+		// ensures vals and bit are 1-indexed
+		for (int i = 1; i <= n; i++) { vals[i].push_back(0); }
+		for (auto [x, y] : todo) {
+			for (int z = x; z <= n; z += z & -z) {
+				if (vals[z].back() != y) { vals[z].push_back(y); }
+			}
+		}
+		for (int i = 1; i <= n; i++) { bit[i].resize(vals[i].size()); }
+		mode = 1;
+	}
+	/** adds t to the point (x, y) */
+	void upd(int x, int y, int t = 1) {
```

Comment:
ok now this might be a tedious task but i think there's a better way clean code-wise to do this

have a constructor that takes a vector of x, y, and t
run `upd` on each of those tuplels and then run the code in `init`
this way there's no need for a weird `mode` variablel

Reply:
fixed I hope?